src/services/ledger/tx-metadata.ts

```diff
@@ -315,6 +315,125 @@ export const LnIntraledgerLedgerMetadata = <
   return { metadata, debitAccountAdditionalMetadata }
 }
 
+export const OnChainTradeIntraAccountLedgerMetadata = ({
+  amountDisplayCurrency,
+  payeeAddresses,
+  sendAll,
+  memoOfPayer,
+}: {
+  amountDisplayCurrency: DisplayCurrencyBaseAmount
+  payeeAddresses: OnChainAddress[]
+  sendAll: boolean
+  memoOfPayer?: string
+}) => {
+  const metadata: AddOnChainTradeIntraAccountLedgerMetadata = {
+    type: LedgerTransactionType.OnChainTradeIntraAccount,
+    pending: false,
+    usd: amountDisplayCurrency,
+    memoPayer: undefined,
+    payee_addresses: payeeAddresses,
+    sendAll,
+  }
+  const debitAccountAdditionalMetadata = {
+    memoPayer: memoOfPayer,
+  }
+  return { metadata, debitAccountAdditionalMetadata }
+}
+
+export const WalletIdTradeIntraAccountLedgerMetadata = <
+  S extends WalletCurrency,
+  R extends WalletCurrency,
+>({
+  paymentFlow,
+  feeDisplayCurrency,
+  amountDisplayCurrency,
+  displayCurrency,
+  memoOfPayer,
+}: {
+  paymentFlow: PaymentFlowState<S, R>
+  feeDisplayCurrency: DisplayCurrencyBaseAmount
+  amountDisplayCurrency: DisplayCurrencyBaseAmount
+  displayCurrency: DisplayCurrency
+  memoOfPayer?: string
+}) => {
+  const {
+    btcPaymentAmount: { amount: satsAmount },
+    usdPaymentAmount: { amount: centsAmount },
+    btcProtocolFee: { amount: satsFee },
+    usdProtocolFee: { amount: centsFee },
+  } = paymentFlow
+
+  const metadata: NewAddWalletIdTradeIntraAccountLedgerMetadata = {
+    type: LedgerTransactionType.WalletIdTradeIntraAccount,
+    pending: false,
+    memoPayer: memoOfPayer,
```

Comment:
we could be consistent on the wording on this one

Reply:
We have `memoPayer`/`memoOfPayer` scatter throughout the codebase that needs to be cleaned up. This could probably be a separate task.